### PR TITLE
Added a new chapter on preparing the VM host

### PR DESCRIPTION
**Pull request by tbazant**
### Description

VM guests need storage and network resource provided by the VM host. This new chapter consolidates existing content and moves it into the libvirt chapter right before guest installation.

### To which product versions do the changes apply?

The first column is to be filled by the requester, the second column is to be filled by the doc team after the changes have been backported to the maintenance branches.

| Code 15     | Backport Done
| ----------  | ---------- 
| [x] 15 SP3  | (`master` only)   
| [ ] 15 SP2  | [ ] 
| [ ] 15 SP1  | [ ] 
| [ ] 15 SP0  | [ ] 

| Code 12     | Backport Done
| ----------  | ---------- 
| [ ] 12 SP5  | [ ] 
| [ ] 12 SP4  | [ ] 
| [ ] 12 SP3  | [ ] 
| [ ] 12 SP2  | [ ]

### CODE REVIEW

File: xml/libvirt_host.xml [hunk 1]
```diff
@@ -0,0 +1,1997 @@
...
+          <imagedata fileref="libvirt_vmm_vnet_ipv4.png" width="45%" format="PNG"/>
+         </imageobject>
+        </mediaobject>
+       </figure>
+      </step>
+      <step>
+       <para>
+        &libvirt; can provide your virtual network with a DHCP server. If you
+        need it, activate <guimenu>Enable DHCPv4</guimenu>, then type the start
+        and end IP address range of assignable addresses.
```
Comment (lproven) [inline xml/libvirt_host.xml:480]:
```suggestion
        and end IP addresses of a range to be made assignable.
```

File: xml/libvirt_host.xml [hunk 2]
```diff
@@ -0,0 +1,1997 @@
...
+     <para>
+      To define a new virtual network without activating it, run
+     </para>
+<screen>&prompt.sudo;virsh net-define <replaceable>VNET_DEFINITION.xml</replaceable></screen>
+     <para>
+      The following examples illustrate definitions of different types of
+      virtual networks.
+     </para>
+     <example xml:id="ex-libvirt-net-nat">
+      <title>NAT Based Network</title>
```
Comment (lproven) [inline xml/libvirt_host.xml:749]:
```suggestion
      <title>NAT-Based Network</title>
```

File: xml/libvirt_host.xml [hunk 3]
```diff
@@ -0,0 +1,1997 @@
...
+      <para>
+       The following configuration allows &vmguest;s outgoing connectivity if
+       it is available on &vmhost;. In the absence of &vmhost; networking, it
```
Comment (lproven) [inline xml/libvirt_host.xml:752]:
```suggestion
       it is available on the &vmhost;. In the absence of &vmhost; networking, it
```

File: xml/libvirt_host.xml [hunk 4]
```diff
@@ -0,0 +1,1997 @@
...
+      <calloutlist>
+       <callout arearefs="vnet-xml-name">
+        <para>
+         The name of the new virtual network.
+        </para>
+       </callout>
+       <callout arearefs="vnet-xml-bridge">
+        <para>
+         The name of the bridge device used to construct the virtual network.
+         When defining a new network with a &lt;forward&gt; mode of "nat" or
```
Comment (lproven) [inline xml/libvirt_host.xml:782]:
```suggestion
         When defining a new network with a &lt;forward&gt; mode of <literal>"nat"</literal> or
```

File: xml/libvirt_host.xml [hunk 5]
```diff
@@ -0,0 +1,1997 @@
...
+         "route" (or an isolated network with no &lt;forward&gt; element),
```
Comment (lproven) [inline xml/libvirt_host.xml:783]:
```suggestion
         <literal>"route"</literal> (or an isolated network with no &lt;forward&gt; element),
```

File: xml/libvirt_host.xml [hunk 6]
```diff
@@ -0,0 +1,1997 @@
...
+         &libvirt; will automatically generate a unique name for the bridge
+         device if none is given.
+        </para>
+       </callout>
+       <callout arearefs="vnet-xml-forward">
+        <para>
+         Inclusion of the &lt;forward&gt; element indicates that the virtual
+         network will be connected to the physical LAN. The
+         <literal>mode</literal> attribute specifies the forwarding method. The
+         most common modes are "nat" (default, network address translation),
```
Comment (lproven) [inline xml/libvirt_host.xml:793]:
```suggestion
         most common modes are <literal>"nat"</literal> (Network Address Translation, the default),
```

File: xml/libvirt_host.xml [hunk 7]
```diff
@@ -0,0 +1,1997 @@
...
+         "route" (direct forwarding to the physical network, no address
```
Comment (lproven) [inline xml/libvirt_host.xml:794]:
```suggestion
         <literal>"route"</literal> (direct forwarding to the physical network, no address
```

File: xml/libvirt_host.xml [hunk 8]
```diff
@@ -0,0 +1,1997 @@
...
+         translation), and "bridge" (network bridge configured outside of
```
Comment (lproven) [inline xml/libvirt_host.xml:795]:
```suggestion
         translation), and <literal>"bridge"<literal> (network bridge configured outside of
```

File: xml/libvirt_host.xml [hunk 9]
```diff
@@ -0,0 +1,1997 @@
...
+       <callout arearefs="vnet-xml-ip">
+        <para>
+         The IP address and netmask for the network bridge.
+        </para>
+       </callout>
+       <callout arearefs="vnet-xml-dhcp">
+        <para>
+         Enable DHCP server for the virtual network, offering IP addresses
+         ranging from the specified <literal>start</literal> and
+         <literal>end</literal> attribute.
```
Comment (lproven) [inline xml/libvirt_host.xml:811]:
```suggestion
         <literal>end</literal> attributes.
```

File: xml/libvirt_host.xml [hunk 10]
```diff
@@ -0,0 +1,1997 @@
...
+   or to attach existing images to the &vmguest;. However, this is not possible
+   when managing &vmguest;s from a remote host. For this reason, &libvirt;
+   supports so called <quote>Storage Pools</quote>, which can be accessed from
+   remote machines.
+  </para>
+
+  <tip>
+   <title>CD/DVD ISO images</title>
+   <para>
+    To be able to access CD/DVD ISO images on the &vmhost; from remote, they
```
Comment (lproven) [inline xml/libvirt_host.xml:1013]: From remote what? From remote hosts, or remote clients, or remote machines in general?

File: xml/libvirt_host.xml [hunk 11]
```diff
@@ -0,0 +1,1997 @@
...
+cd /etc/libvirt/qemu
+for FILE in *.xml; do
+  basename $FILE .xml
+  xsltproc $SSHEET $FILE
+done</screen>
+     </step>
+    </procedure>
+   </sect3>
+   <sect3 xml:id="sec-libvirt-storage-virsh-start-pools">
+    <title>Starting, Stopping and Deleting Pools</title>
```
Comment (lproven) [inline xml/libvirt_host.xml:1274]:
```suggestion
    <title>Starting, Stopping, and Deleting Pools</title>
```

File: xml/libvirt_host.xml [hunk 12]
```diff
@@ -0,0 +1,1997 @@
...
+   </para>
+  </warning>
+
+  <sect2 xml:id="sec-libvirt-storage-virsh">
+   <title>Managing Storage with <command>virsh</command></title>
+   <para>
+    Managing storage from the command line is also possible by using
+    <command>virsh</command>. However, creating storage pools is currently not
+    supported by &suse;. Therefore, this section is restricted to documenting
+    functions like starting, stopping and deleting pools and volume management.
```
Comment (lproven) [inline xml/libvirt_host.xml:1207]:
```suggestion
    functions such as starting, stopping, and deleting pools, and volume management.
```

File: xml/libvirt_host.xml [hunk 13]
```diff
@@ -0,0 +1,1997 @@
...
+     also list inactive pools, add the option <option>--all</option>:
+    </para>
+<screen>&prompt.user;virsh pool-list --details</screen>
+    <para>
+     Details about a specific pool can be obtained with the
+     <literal>pool-info</literal> subcommand:
+    </para>
+<screen>&prompt.user;virsh pool-info <replaceable>POOL</replaceable></screen>
+    <para>
+     Volumes can only be listed per pool by default. To list all volumes from a
```
Comment (lproven) [inline xml/libvirt_host.xml:1227]:
```suggestion
     By default, volumes can only be listed per pool. To list all volumes from a
```

File: xml/libvirt_host.xml [hunk 14]
```diff
@@ -0,0 +1,1997 @@
...
+  basename $FILE .xml
+  xsltproc $SSHEET $FILE
+done</screen>
+     </step>
+    </procedure>
+   </sect3>
+   <sect3 xml:id="sec-libvirt-storage-virsh-start-pools">
+    <title>Starting, Stopping and Deleting Pools</title>
+    <para>
+     Use the <command>virsh</command> pool subcommands to start, stop or delete
```
Comment (lproven) [inline xml/libvirt_host.xml:1276]:
```suggestion
     Use the <command>virsh</command> pool subcommands to start, stop, or delete
```

File: xml/libvirt_host.xml [hunk 15]
```diff
@@ -0,0 +1,1997 @@
...
+     </callout>
+     <callout arearefs="co-vol-create-as-capacity">
+      <para>
+       Size of the image, in this example 12 gigabytes. Use the suffixes k, M,
+       G, T for kilobyte, megabyte, gigabyte, and terabyte, respectively.
+      </para>
+     </callout>
+     <callout arearefs="co-vol-create-as-format">
+      <para>
+       Format of the volume. &suse; currently supports <literal>raw</literal>,
```
Comment (lproven) [inline xml/libvirt_host.xml:1366]:
```suggestion
       Format of the volume. &suse; currently supports <literal>raw</literal>
```

Comment (lproven) [reply xml/libvirt_host.xml:1366]: I think we don't need an Oxford comma when there are only 2 items.

File: xml/libvirt_host.xml [hunk 16]
```diff
@@ -0,0 +1,1997 @@
...
+       and <literal>qcow2</literal>.
+      </para>
+     </callout>
+     <callout arearefs="co-vol-create-as-alloc">
+      <para>
+       Optional parameter. By default <command>virsh </command>creates a sparse
```
Comment (lproven) [inline xml/libvirt_host.xml:1372]:
```suggestion
       Optional parameter. By default, <command>virsh</command> creates a sparse
```

File: xml/libvirt_host.xml [hunk 17]
```diff
@@ -0,0 +1,1997 @@
...
+      </informalfigure>
+     </step>
+     <step>
+      <para>
+       Specify the required details in the following window. The data that
+       needs to be entered depends on the type of pool you are creating:
+      </para>
+      <variablelist>
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>dir</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1586]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>dir</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 18]
```diff
@@ -0,0 +1,1997 @@
...
+          <listitem>
+           <para>
+            <guimenu>Target Path</guimenu>: Specify an existing directory.
+           </para>
+          </listitem>
+         </itemizedlist>
+        </listitem>
+       </varlistentry>
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>disk</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1598]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>disk</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 19]
```diff
@@ -0,0 +1,1997 @@
...
+           <para>
+            <guimenu>Build Pool</guimenu>: Activating this option formats the
+            device. Use with care&mdash;all data on the device will be lost!
+           </para>
+          </listitem>
+         </itemizedlist>
+        </listitem>
+       </varlistentry>
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>fs</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1637]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>fs</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 20]
```diff
@@ -0,0 +1,1997 @@
...
+         <listitem>
+          <para>
+           <guimenu>Source Path</guimenu>: ...
+          </para>
+         </listitem>
+        </itemizedlist>
+       </listitem>
+      </varlistentry>-->
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>iscsi</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1688]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>iscsi</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 21]
```diff
@@ -0,0 +1,1997 @@
...
+          <listitem>
+           <para>
+            <guimenu>Source Path</guimenu>: The iSCSI target name (IQN).
+           </para>
+          </listitem>
+         </itemizedlist>
+        </listitem>
+       </varlistentry>
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>logical</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1723]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>logical</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 22]
```diff
@@ -0,0 +1,1997 @@
...
+           <para>
+            <guimenu>Build Pool</guimenu>: Only activate when creating a new
+            volume group.
+           </para>
+          </listitem>
+         </itemizedlist>
+        </listitem>
+       </varlistentry>
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>mpath</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1750]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>mpath</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 23]
```diff
@@ -0,0 +1,1997 @@
...
+         <listitem>
+          <para>
+           <guimenu>Source Name</guimenu>: ...
+          </para>
+         </listitem>
+        </itemizedlist>
+       </listitem>
+      </varlistentry>-->
+       <varlistentry>
+        <term><emphasis role="bold">Type</emphasis><guimenu>scsi</guimenu></term>
```
Comment (lproven) [inline xml/libvirt_host.xml:1807]:
```suggestion
        <term><emphasis role="bold">Type</emphasis> <guimenu>scsi</guimenu></term>
```

File: xml/libvirt_host.xml [hunk 24]
```diff
@@ -0,0 +1,1997 @@
...
+         </listitem>
+        </itemizedlist>
+       </listitem>
+      </varlistentry>-->
+      </variablelist>
+      <note>
+       <title>File Browsing</title>
+       <para>
+        Using the file browser by clicking <guimenu>Browse</guimenu> is not
+        possible when operating from remote.
```
Comment (lproven) [inline xml/libvirt_host.xml:1847]: Again: when operating from **a** remote _what_?

If this is not know, then I suggest: 

... is not possible when operating remotely.

**Review by lproven — CHANGES_REQUESTED** [review]

File: xml/libvirt_host.xml [hunk 25]
```diff
@@ -0,0 +1,1997 @@
...
+        In the <guimenu>Address</guimenu> tab, specify networking details such
+        as DHCP/static IP address, subnet mask or host name.
+       </para>
+       <para>
+        Using <guimenu>Dynamic Address</guimenu> is only useful when also
+        assigning a device to a bridge that is connected to a DHCP server.
+       </para>
+       <para>
+        If you intend to create a virtual bridge that has no connection to a
+        real Ethernet device, use <guimenu>Statically assigned IP
```
Comment (tbazant) [inline xml/libvirt_host.xml:114]:
```suggestion
        real network device, use <guimenu>Statically assigned IP
```

File: xml/libvirt_host.xml (hunk 10 above)
Comment (tbazant) [inline xml/libvirt_host.xml:1013]:
```suggestion
    To be able to access CD/DVD ISO images on the &vmhost; from remote clients, they
```

File: xml/libvirt_host.xml (hunk 24 above)
Comment (tbazant) [inline xml/libvirt_host.xml:1847]:
```suggestion
        possible when operating remotely.
```

**Review by tbazant — COMMENTED** [review]
Added a few suggestions as hinted by @lproven

**Review by aginies — APPROVED** [review]
Sounds ok, i read quickly the end, as the changes are mostly on the top.

**Pull request merged**